akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala

```diff
@@ -237,4 +237,10 @@ private object PoolSlot {
         context.stop(self)
     }
   }
+
+  final class UnexpectedDisconnectException(msg: String, cause: Throwable) extends RuntimeException(msg, cause) {
+    def this(msg: String) {
```

Comment:
yep, will use that. I so rarely write secondary constructors that I'm always confused about syntax of them ;)